cpp/src/arrow/datum.cc

```diff
@@ -20,6 +20,7 @@
 #include <cstddef>
 #include <memory>
 #include <sstream>
+#include <variant>
```

Comment:
This shouldn't be required, should it?

Reply:
`std::visit` for variants is defined there, so our IWYU policy does recommend this

Reply:
Hmm. `variant` is already included by `datum.h`, is it?

Reply:
IWYU need variant for visit，the variant in datum.h  only marked variant